### PR TITLE
[Geant4] Code adjustment for new Geant4 version

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:
Changes to get new Geant4 10.6 external for the cmsdist Geant4 branch.
See cms-externals/geant4#42 , cms-sw/cmssw#28626 and cms-sw/cmssw#28629. This PR joins 28626 and 28629 + the requirements discussed in them. 

#### PR validation:

Builds for both cmsdist/master (production "old" geant) and cmsdist/geant4 (10.6 geant)

The previous two cmssw PRs discussion suggests to use int counters to avoid the original build error and from other concerns. This PR builds with both master of cmdist and geant4 branch (works with both geant versions)

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
this looks good to go in master IB. Once it is merged then we will merge cmsdist geant4 changes cms-sw/cmsdist#5424

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

File: SimG4CMS/ShowerLibraryProducer/src/HcalForwardAnalysis.cc [hunk 1]
```diff
@@ -129,9 +129,10 @@ void HcalForwardAnalysis::setPhotons(const EndOfEvent* evt) {
   std::vector<HFShowerPhoton> LongFiberPhotons;
   LongFiberPhotons.clear();
   ShortFiberPhotons.clear();
+  int thehc_entries = theHC->entries();
   if (idHC >= 0 && theHC != nullptr) {
-    std::cout << "FiberhitSize " << theHC->entries() << std::endl;
-    for (j = 0; j < theHC->entries(); j++) {
+    std::cout << "FiberhitSize " << thehc_entries << std::endl;
```
Comment (fabiocos) [inline SimG4CMS/ShowerLibraryProducer/src/HcalForwardAnalysis.cc]: cout is not ideal, but this code is likely not part of official workflows meant to run in multithreaded mode

**Comment by fabiocos** [comment]
+1

technical update

**Pull request merged**